service/controller/v2/key/key.go

```diff
@@ -209,6 +209,17 @@ func ToCustomObject(v interface{}) (providerv1alpha1.AzureConfig, error) {
 	return customObject, nil
 }
 
+func ToNodeCount(v interface{}) (int, error) {
```

Comment:
This should be rather `func NodeCount`

Reply:
Why? We have the `To` prefix as a pretty common thing for everything that takes an interface value and returns something concrete.

Reply:
Other functions in key package don't have `To` prefix. This prefix indicates for me the function is on conversion which it isn't.

Reply:
One of the very first functions and a reason why we invented the key package at all was `ToCustombject`.

Reply:
In fact we fist had extractors and the decided to move `ToCustombject` to the `key` package.

But I understand what is weird here. Normally it would look like:

```
func NodeCount(customObject providerv1alpha1.AzureConfig) int {
	nodeCount := len(customObject.Spec.Cluster.Masters) + len(customObject.Spec.Cluster.Workers)

	return nodeCount
}
```

That's what confused me and I didn't realize. Why this one does both conversion and extraction?

Reply:
Because we use it to configure the `statusresource`, which is provider independent. See https://github.com/giantswarm/statusresource/blob/4193e00bf9e1bfcf2fa4a38480589bdf0e7fb6b5/resource.go#L17.

Reply:
Ack 👍